src/app.rs

```diff
@@ -451,6 +457,23 @@ where
     }
 }
 
+fn extract_auth_header(headers: HeaderMap) -> Option<BTreeMap<String, String>> {
```

Comment:
Looks like there isn't a reason to take ownership of `headers` here. Changing this to `headers: &HeaderMap` should avoid the need to `.clone()` the `HeaderMap` above.

Reply:
Fixed